src/supplier/supplier.repository.ts

```diff
@@ -12,8 +12,31 @@ export class SupplierRepository {
     this.repository = this.dataSource.getRepository(Supplier);
   }
 
-  findAllSupplier(): Promise<Supplier[]> {
-    return this.repository.find();
+  async findAllSupplier(
+    page: number,
+    limit: number,
+  ): Promise<{
+    dataSupplier: Supplier[];
+    total: number;
+    currentPage: number;
+    lastPage: number;
+  }> {
+    const [data, total] = await this.repository.findAndCount({
```

Comment:
add search and filter

Reply:
done create search filter for nama, alamat and nomor telepon